File: 4.8-bc/py/stage-2-feature-first/app/features/catalog/service.py
"""Catalog feature — list products."""
from __future__ import annotations

from app.features.catalog.model import Product
from app.features.catalog.repository import Repository


class Service:
    def __init__(self, repo: Repository) -> None:
        self._repo = repo

    async def list(self) -> list[Product]:
        return await self._repo.list()

File: 4.8-bc/ts/stage-2-feature-first/src/features/catalog/service.ts
// Catalog feature — service: продукти, категорії.
import type { Product } from "./model.js";
import type { Repository } from "./repository.js";

export class Service {
  constructor(private readonly repo: Repository) {}

  list(): Promise<Product[]> {
    return this.repo.list();
  }
}
